pyloopkit/pyloop_parser.py

```diff
@@ -102,13 +102,18 @@ def get_insulin_data(
     ]
     values = []
     for i in range(0, len(data)):
+        value = float(data[i].get("value"))
+        if 'deliveredUnits' in data[i].keys() and data[i].get("deliveredUnits") != 'nil':
+            value = float(data[i].get("deliveredUnits"))
+
+
         values.append(
             convert_to_correct_units(
```

Comment:
The value in `deliveredUnits` is always already in `U`, so it should never be converted.

Reply:
fixed